extensions/fairifier-rdf-extension/dtls/fairifier/FileSystemRdfSkeletonImpl.java

```diff
@@ -1,89 +1,96 @@
-                SkeletonMetadata metadata = mapper.readValue(new File(element.getParent() + File.separator + name + "metadata.skeleton.json"), SkeletonMetadata.class);
-                out.add(new String(Files.readAllBytes(element)));
-            } catch (IOException e) {
-                throw new UncheckedIOException(e);
+    public List<String> listModels()  throws IOException {
+        ArrayList<String> outList = new ArrayList<String>();
+        List<String[]> files = Files.list(SAVELOCATION).filter(Files::isRegularFile).map(this::getStringAndFiletype).collect(Collectors.toList());;
+        for (String[] o : files) {
+            if (o != null) { 
+                outList.add(o[0]);
             }
-        });
-        return out;
+        }
+        return outList;
     }
     
     @Override
-    public void saveModel(String json, String fileType, String projectId) {
-        try {
-            BufferedWriter out = new BufferedWriter(new FileWriter(SAVELOCATION.toString() + File.separator + projectId + ".skeleton.json"));
-            out.write(json);
-            out.close();
-            ObjectMapper mapper = new ObjectMapper();
+    public void saveModel(String json, String fileType, String projectId)  throws IOException {
+        Path fileLocation = Paths.get(SAVELOCATION.toString() + File.separator + projectId + ".skeleton.json");
+        Files.write(fileLocation, json.getBytes(StandardCharsets.UTF_8));
+        try (BufferedWriter writer = Files.newBufferedWriter(fileLocation)) {
             SkeletonMetadata metadata = new SkeletonMetadata();
             metadata.setFileType(fileType);
-            mapper.writeValue(new File(SAVELOCATION.toString() + File.separator  + projectId + ".metadata.skeleton.json"), metadata);
+            mapper.writeValue(writer, metadata);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
     @Override
-    public String loadModel(String projectId) {
-        Stream<Path> files = Files.list(SAVELOCATION);
-        String out;
-        files.forEach(element -> {
-            String name = element.getFileName().toString().split(".")[0];
-            if(name.equals(projectId)) {
-                try {
-                    out = new String(Files.readAllBytes(element));
-                } catch (IOException e) {
-                    throw new UncheckedIOException(e);
-                }
+    public String loadModel(String projectId)  throws IOException {
+        java.util.Optional<String[]> selected = Files.list(SAVELOCATION)
+                .filter(Files::isRegularFile)
+                .filter(path -> {
+                    String filename = path.getFileName().toString();
+                    String[] tokens = filename.split("\\.");
+                    return tokens.length == 3 && tokens[0].equals(projectId);
+                })
+                .map(this::getStringAndFiletype)
+                .findFirst();
+        return selected.get()[1];
```

Comment:
The `selected.get()[1]` can be incorporated into the stream logic as well, no need to exit the stream logic for that.